comit/src/expiries/config.rs

```diff
@@ -0,0 +1,289 @@
+//! This module provides functionality for calculating the duration of actions
+//! required to determine the transition period from one swap state to the next.
+
+use std::f64::consts;
+use time::{prelude::*, Duration};
+
+// TODO: From somewhere within the system we need to return to the
+// user a transaction fee to use for each of the transactions (deploy,
+// fund, refund, redeem). In this module we assume the fee is set to
+// the suggested amount. We rely on this assumption for the
+// confirmation time calculations in this module to be correct.
+//
+// For Ethereum:
+//
+// gas limit: blockchain-contracts already implements gas limit functionality
+//            for each Ethereum transaction type.
+// gas price: current gas price can be learned from the Ethereum blockchain,
+//            for example from geth or infura.
+// modifier:  We could use a pre-configured modifier, as is widely done in the
+//            Ethereum ecosystem for slow, medium, fast.
+//
+// For Bitcoin:
+//
+// With a configured 'include within N blocks' we can use bitcoind via
+// bitcoin-cli, by calling:
+//
+//  `bitcoin-cli estimatesmartfee N_BLOCKS`
+//
+// CAVEAT: The fee estimator relies on an active mempool with good uptime to
+// watch fee activity on the network, that in turn implies that our calculations
+// herein are only as good as the fee estimator of the Bitcoin connector.
+
+const BITCOIN_BLOCK_TIME_SECS: usize = 600; // 10 minutes, average Bitcoin block time.
+const ETHEREUM_BLOCK_TIME_SECS: usize = 20; // Conservative Ethereum block time.
```

Comment:
Why using a conservative and not average/target block time?

Reply:
The only information I could find suggested that Ethereum average block time is typically between 10 and 20 seconds. Since this is so small (compared to Bitcoin) I figured just take a conservative position and use 20.

Reply:
Oh, I thought it was officially 17s.